### PR TITLE
Add the evalpoly function to the docs

### PR DESCRIPTION
**Pull request by ararslan**
Also fix some docstring formatting for the function.

### CODE REVIEW

**Review by fredrikekre — APPROVED** [review]
Seems a bit excessive to have 4 docstrings for the same function.

**Comment by ararslan** [comment]
Yeah I agree. I can push a commit that consolidates them.

**Comment by ararslan** [comment]
Also needs a compat annotation.

File: base/math.jl [hunk 1]
```diff
@@ -83,10 +83,20 @@ end
...
+    evalpoly(x, p)
+
 Evaluate the polynomial ``\\sum_k p[k] x^{k-1}`` for the coefficients `p[1]`, `p[2]`, ...;
-that is, the coefficients are given in ascending order by power of `x`. This function
-generates efficient code using Horner's method with loops unrolled at compile time.
+that is, the coefficients are given in ascending order by power of `x`.
+Loops are unrolled at compile time if the number of coefficients is statically known, i.e.
+when `p` is a `Tuple`.
+This function generates efficient code using Horner's method if `x` is real, or using
+a Goertzel-like algorithm if `x` is complex.
```
Comment (fredrikekre) [inline base/math.jl]: Maybe make this a footnote
```suggestion
a Goertzel-like [^DK62] algorithm if `x` is complex.

[^DK62]: Donald Knuth, Art of Computer Programming, Volume 2: Seminumerical Algorithms, Sec. 4.6.4.
```

File: base/math.jl [hunk 2]
```diff
@@ -83,10 +83,20 @@ end
...
+The Goertzel-like algorthim is described in Knuth's _Art of Computer Programming_,
```
Comment (fredrikekre) [inline base/math.jl]:
```suggestion
```

File: base/math.jl [hunk 3]
```diff
@@ -83,10 +83,20 @@ end
...
+Volume 2: Seminumerical Algorithms, Sec. 4.6.4.
```
Comment (fredrikekre) [inline base/math.jl]:
```suggestion
```

File: base/math.jl (hunk 1 above)
Comment (ararslan) [reply base/math.jl]: Whoa, TIL. Cool!

**Comment by ararslan** [comment]
I guess I could move the examples from the deleted docstrings to the new one as well.

**Comment by ararslan** [comment]
Actually, meh, they're not particularly informative over what's already there.

**Pull request merged**